src/lib.rs

```diff
@@ -40,12 +40,21 @@ mod nodes;
 // TODO(jmmv): Make configurable via a flag and store inside SandboxFS.
 pub const TTL: Timespec = Timespec { sec: 60, nsec: 0 };
 
-/// An error indicating that a path has to be absolute but isn't.
-#[derive(Debug, Fail)]
-#[fail(display = "path {:?} is not absolute", path)]
-pub struct PathNotAbsoluteError {
-    /// The path that caused this error.
-    pub path: PathBuf,
+/// An error indicating that a mapping specification (coming from the command line or from a
+/// reconfiguration operation) is invalid.
+#[derive(Debug, Eq, Fail, PartialEq)]
+pub enum MappingError {
+    /// A path was required to be absolute but wasn't.
+    #[fail(display = "path {:?} is not absolute", path)]
```

Comment:
`#[fail(display = "path {} is not absolute", path.display())]`

Reply:
Too good to be true :(

From https://boats.gitlab.io/failure/derive-fail.html:

```
Note that unlike code that would appear in a method, this does not use something like self.code or self.message; it just uses the field names directly. This is because of a limitation in Rust's current attribute syntax. As a result, you can only interpolate fields through the derivation; you cannot perform method calls or use other arbitrary expressions.
```

so the call to `display()` doesn't work.